0000-deb-repo-sync-gpg-check.md

```diff
@@ -0,0 +1,178 @@
+- Feature Name: deb_repo_sync_gpg_check
+- Start Date: 12-06-2019
+- RFC PR: (leave this empty)
+
+# Summary
+
+Verify GPG signature of Debian repositories when synchronizing.
+
+# Motivation
+
+- Why are we doing this?
+- What use cases does it support?
+- What is the expected outcome?
+
+Uyuni can synchronize Debian repositories but currently it doesn't verify the GPG signature of the repo metadata.
+In order to import the metadata in a secure way Uyuni must be able to verify the GPG signature of that metadata.
+
+# Detailed design
+
+## Database changes
+
+A new table is needed to store the association of a GPG key to a repository:
+```
+rhnContentSourceGpg
+====================
+content_source_id   number not null foreign key rhnContentSource(id)
+gpg_key_id          number not null foreign key rhnCryptoKey(id)
+created             timestamp
+modified            timestamp
+
+```
+
+## Repo signature checking
+
+The GPG verification should be optional but a warning should be shown to the the user in case he's not providing information about the GPG key used to verify the metadata.
```

Comment:
I would not say "is optional", instead I would say, "can be disabled". The default should always be "check it".
Making it insecure is an option an admin has but we do not recommend it.

Reply:
Ok, I made verification mandatory unless explicitly disabled.